support-workers/src/main/scala/com/gu/support/workers/lambdas/CreateZuoraSubscription.scala

```diff
@@ -199,6 +189,106 @@ object ZuoraSubscriptionCreator {
 
 }
 
+class NextState(
+  state: CreateZuoraSubscriptionState,
+  accountNumber: ZuoraAccountNumber,
+  subscriptionNumber: ZuoraSubscriptionNumber,
+) {
+
+  def getEmailState(
+    paymentOrRedemptionData: Either[PaymentMethodWithSchedule, RedemptionData],
+    maybeDigitalSubscriptionGiftCreationDetails: Option[DigitalSubscriptionGiftCreationDetails],
+  ): SendAcquisitionEventState =
+    SendAcquisitionEventState(
+      analyticsInfo = state.analyticsInfo,
+      sendThankYouEmailState = getProductSpecificState(
+        paymentOrRedemptionData,
+        maybeDigitalSubscriptionGiftCreationDetails
+      ),
+      acquisitionData = state.acquisitionData
+    )
+
+  val Purchase = Left
+  type Redemption = Right[PaymentMethodWithSchedule, RedemptionData]
+
+  // scalastyle:off cyclomatic.complexity
```

Comment:
Do we still need to disable scalastyle here?

Reply:
yes otherwise it complains 
![image](https://user-images.githubusercontent.com/7304387/97598490-f2279500-19fe-11eb-9eb0-68737642f7a2.png)
although I can't work out why 13 and not 8.  I can only assume it's how it generates the bytecode for the match statement.
There is a "countCases: Y/N" option in scalastyle, but it's only  in the dev branch http://www.scalastyle.org/rules-dev.html#parameters-13:~:text=countCases,-Count

Reply:
re 13 vs 8, looks like scalastyle bugs really.

looks like it searches for IF, WHILE, DO, FOR, CASE and || / &&
https://github.com/scalastyle/scalastyle/blob/ec14399543d2d5ccf93c3713aa5df21793844791/src/main/scala/org/scalastyle/scalariform/CyclomaticComplexityChecker.scala#L43
So I guess the if guard on each case has confused it into thinking it's another code path.
Then it adds one for the default case (which doesn't exist as it's covered by the `case _`.